python/tests/server/test_worker.py

```diff
@@ -351,6 +351,57 @@ def test_async_predictor_on_python_3_10_or_older_raises_error(worker):
     )
 
 
+@uses_worker(
+    "setup_async", max_concurrency=1, min_python=(3, 11), is_async=True, setup=False
+)
+def test_setup_async(worker: Worker):
+    fut = worker.setup()
+    setup_result = Result()
+    setup_sid = worker.subscribe(setup_result.handle_event)
+
+    # with pytest.raises(FatalWorkerException):
+    fut.result()
+    worker.unsubscribe(setup_sid)
+
+    assert setup_result.stdout_lines == [
+        "setup starting...\n",
+        "download complete!\n",
+        "setup complete!\n",
+    ]
+
+    predict_result = Result()
+    predict_sid = worker.subscribe(predict_result.handle_event, tag="p1")
+    worker.predict({}, tag="p1").result()
+
+    assert predict_result.done
+    assert predict_result.output == "output"
+    assert predict_result.stdout_lines == ["running prediction\n"]
+
+    worker.unsubscribe(predict_sid)
+
```

Comment:
can we also have a test that uses the `"async_setup_uses_same_loop_as_predict"` fixture? (It already exists in the code tree but nothing currently references it, and some models rely on setup and predict using the same event loop.